app/attributes/attributecontroller.cpp

```diff
@@ -632,6 +632,15 @@ void AttributeController::acquireId()
     emit commitFailed();
   }
 
+  //
+  // We need to trigger the recalucation of the form because QgsVectorLayer::featureAdded
+  // is called before commit is finished. This can cause issues with evaluation of validations
+  // when old feature with FID_IS_NEW is not yet removed from the layer and the new feature
+  // with valid fid is already added (unique constraints are failing as it detects two features
+  // with the same values). We thus manually trigger the recalculation here.
+  //
+  recalculateDerivedItems();
```

Comment:
I don't fully grasp what is happening in this function, but why is `recalculateDerivedItems`  called without arguments when the function signature got two parameters ?

Reply:
Hi @ValentinBuira, both arguments have default value, see https://github.com/MerginMaps/mobile/blob/052b5b663e3ad6095995f58adb16f40a82c302ae/app/attributes/attributecontroller.h#L195 you do not need to pass values if you are happy with their defaults.